backend/sites/src/app/services/snapshot/snapshot.service.spec.ts

```diff
@@ -534,6 +534,79 @@ describe('SnapshotService', () => {
     });
   });
 
+  describe('getMostRecentSnapshot', () => {
+    it('should return a single snapshot for a given userId and siteId', async () => {
+      const userId = '1';
+      const siteId = '1';
+      const res: Snapshots = {
+        id: 1,
+        userId: '1',
+        siteId: '1',
+        transactionId: '1',
+        whenCreated: new Date(2024, 8, 28),
+        whoCreated: 'ABC',
+        whenUpdated: new Date(2024, 8, 28),
+        whoUpdated: 'ABC',
+        site: sampleSites[0],
+        snapshotData: {
+          sitesSummary: sampleSites[0],
+          documents: null,
+          events: null,
+          eventsParticipants: null,
+          landHistories: null,
+          profiles: null,
+          siteAssociations: null,
+          subDivisions: null,
+          siteParticipants: null,
+        },
+      };
+      jest
+        .spyOn(snapshotRepository, 'findOne')
+        .mockResolvedValueOnce(res as Snapshots);
+      const snapshot = await service.getMostRecentSnapshot(siteId, userId);
+
+      expect(snapshot).toEqual(res);
+      expect(snapshotRepository.findOne).toHaveBeenCalledTimes(1);
+      expect(snapshotRepository.findOne).toHaveBeenCalledWith({
+        where: { siteId, userId },
+        order: { whenCreated: 'DESC' },
+      });
+    });
+
+    it('should throw an error if repository throws an error', async () => {
+      const userId = '1';
+      const siteId = '1';
+      const errorMessage = 'Failed to retrieve the most recent snapshot';
+      jest
+        .spyOn(snapshotRepository, 'findOne')
+        .mockRejectedValueOnce(new Error(errorMessage));
+
+      await expect(
+        service.getMostRecentSnapshot(siteId, userId),
+      ).rejects.toThrow(errorMessage);
+      expect(snapshotRepository.findOne).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('when there are no results', () => {
+    it('returns null', async () => {
```

Comment:
This might be redundant but I think it's useful to assert that typeorm returning a null result will not cause an error in the snapshot service.